.github/workflows/tests.yaml

```diff
@@ -1,15 +1,36 @@
 name: Tests
-on: # Run on all pushes and pull requests
+on: # Run on all pushes and pull requests. Eventually can run certain jobs on commit.
   push:
     branches:
       - main
   pull_request:
+
 jobs:
-  build:
+  test-base:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11"]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+      - name: Install dependencies
+        run: poetry install
+      - name: Run base tests
+        run: make test-base
+
+  test-iso:
+    needs: [test-base]
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.10", "3.11"]
+        python-version: ["3.11"]
+        iso: [caiso, ercot, isone, miso, nyiso, pjm, spp, ieso, eia, misc]
```

Comment:
Not sure that `misc` should be part of the iso tests. `test_gridstatus.py` and `test_lmp_config.py` seem more like base tests.

Reply:
I'm with you, I didn't intend to have a `misc` category and they originally were in base tests, but it seems they have some external or source-specific dependencies currently. I figured they would be the first low hanging fruit to pick in unpinning external dependencies and setting up VCR a bit more, which I was going to dive into tomorrow. Ideally the misc category only lasts like 24 hours haha

Reply:
Sounds good. Doesn't need to be fixed in this PR.